File: machine/chatbot/node/create_model.js
const tf = require("@tensorflow/tfjs-node");
const { split_sentences, words_to_binary } = require("./de_contract");
const fs = require("fs");

const train_model = async (dataset) => {
  const { words, classes, documents } = split_sentences(dataset);

  const { train_x, train_y } = words_to_binary(words, classes, documents);
  fs.writeFile("words.json", JSON.stringify(words), (err) => {
    if (err) {
      console.log(err);
    }
  });
  fs.writeFile("classes.json", JSON.stringify(classes), (err) => {
    if (err) {
      console.log(err);
    }
  });
  const model = tf.sequential();
  model.add(
    tf.layers.dense({
      units: 128,
      inputShape: [train_x[0].length],
      activation: "relu",
    })
  );
  model.add(tf.layers.dropout({ rate: 0.5 }));
  model.add(
    tf.layers.dense({
      units: 64,
      activation: "relu",
    })
  );
  model.add(tf.layers.dropout({ rate: 0.5 }));
  model.add(
    tf.layers.dense({
      units: train_y[0].length,
      activation: "softmax",
    })
  );
  model.compile({
    loss: "categoricalCrossentropy",
    optimizer: "adam",
    metrics: ["accuracy"],
  });

  const history = await model.fit(tf.tensor(train_x), tf.tensor(train_y), {
    epochs: train_x[0].length,
    batchSize: 15,
  });

  model.save("file://", history);
};

module.exports = train_model;

File: machine/sentiment analysis/python/create_model.py
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout
import json
from de_contract import *


def train_model(dataset):
    words, classes, documents = split_sentences(dataset)

    train_x, train_y = words_to_binary(words, classes, documents)

    with open("words.json", "w") as f:
        json.dump(words, f)

    with open("classes.json", "w") as f:
        json.dump(classes, f)

    model = Sequential()
    model.add(Dense(128, input_shape=(len(train_x[0]),), activation="relu"))
    model.add(Dropout(0.5))
    model.add(Dense(64, activation="relu"))
    model.add(Dropout(0.5))
    model.add(Dense(len(train_y[0]), activation="softmax"))

    model.compile(loss="categorical_crossentropy",
                  optimizer="adam", metrics=["accuracy"])

    history = model.fit(np.array(train_x), np.array(
        train_y), epochs=100, batch_size=15)

    model.save("model.h5", history)